src/macrofier/engine.mjs

```diff
@@ -486,8 +488,66 @@ function insertTableofContents(content) {
   return content
 }
 
+const isEnumType = x => x.type !== 'undefined' && x.type === 'string' && Array.isArray(x.enum)
+
+const getProperties = x => {
+   return Array.isArray(x.properties) ? x.properties[0] : x.properties
+}
+
+const isEnumProperties = schema => compose(
+    getProperties,
+    filter(enm => enm),
+    map(filter(enm => enm)),
+    map(props => props.map(([k, v]) => ((v.type === 'object') ? isEnumProperties(v) : ((v.type === 'array') ? isEnumType(v.items[0] ? v.items[0] : v.items): isEnumType(v))))),
+    map(Object.entries),
+    filter(schema => isObject(schema))
+)(schema)
+
+const getEnumProperties = schema => compose(
+    getProperties,
+    filter(enm => enm),
+    map(filter(isEnumType)),
+    map(props => props.map(([k, v]) => {
+      let enm = v
+      if (isEnumType(v) == true) {
+        enm = Object.assign({}, v)
+        enm.title = k
+      } else if (v.type === 'object') {
+        enm = getEnumProperties(v)
+      } else if (v.type === 'array') {
+        enm = Object.assign({}, (v.items[0] ? v.items[0] : v.items))
+        enm.title = k
+      }
+      return enm
+    })),
+    map(Object.entries),
+    filter(schema => isObject(schema))
+)(schema)
+
+const convertEnumTemplate = (sch, templateName, templates) => {
+  const template = getTemplate(templateName, templates).split('\n')
+  let schema = isEnumType(sch) ? sch : getEnumProperties(sch)
+  for (var i = 0; i < template.length; i++) {
+    if (template[i].indexOf('${key}') >= 0) {
+      template[i] = schema.enum.map(value => {
+        const safeName = value.split(':').pop().replace(/[\.\-]/g, '_').replace(/\+/g, '_plus').replace(/([a-z])([A-Z0-9])/g, '$1_$2').toUpperCase()
+        return template[i].replace(/\$\{key\}/g, safeName)
+                          .replace(/\$\{value\}/g, value)
+      }).join('\n')
+      if (!templateName.includes("enum_conversion")) {
```

Comment:
What's this code doing?

Reply:
excluding the last line comma replacement for enumconversion case, since it need last comma also

Reply:
Got it... let's get this PR merged. at some point I'd like to add a ${delimiter}${end.delimiter} macro, that we can wrap around the comma in the template, and then just remove any text between those macros for the last item.

No need to do it now.